pkg/internal/vault/vault_test.go

```diff
@@ -315,6 +321,114 @@ func TestSetToken(t *testing.T) {
 			expectedToken: "my-roleapp-token",
 			expectedErr:   nil,
 		},
+
+		"if kubernetes role auth set but reference secret doesn't exist return error": {
+			issuer: gen.Issuer("vault-issuer",
+				gen.SetIssuerVault(v1alpha1.VaultIssuer{
+					CABundle: []byte(testCertBundle),
+					Auth: v1alpha1.VaultAuth{
+						Kubernetes: v1alpha1.KubernetesAuth{
+							Role: "kube-vault-role",
+							SecretRef: v1alpha1.SecretKeySelector{
+								LocalObjectReference: v1alpha1.LocalObjectReference{
+									Name: "secret-ref-name",
+								},
+								Key: "my-kube-key",
+							},
+						},
+					},
+				}),
+			),
+			fakeLister: listers.FakeSecretListerFrom(listers.NewFakeSecretLister(),
+				listers.SetFakeSecretNamespaceListerGet(nil, errors.New("secret does not exists")),
+			),
+			fakeClient:    vaultfake.NewFakeClient(),
+			expectedToken: "",
+			expectedErr:   errors.New("error reading Kubernetes service account token from secret-ref-name: secret does not exists"),
```

Comment:
Out of interest, how are consumers of this package detecting that building a client failed due to dependent resources not existing, and not some other transient error? I assume we're not, and just relying on controllers backing off in this case?

Reply:
Yes, we're only relying on backing off atm